gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FlowSpec.java

```diff
@@ -146,14 +150,13 @@ public static FlowSpec.Builder builder(URI catalogURI, Properties flowProps) {
    * @param key
    * @param value
    */
-  public synchronized void addProperty(String key, String value) {
+  public synchronized void addProperty(String key, Object value) {
```

Comment:
this could obscure errors.  I'd be more comfortable w/ an overloaded `addProperty(String, long)` method - would that compile and type check?

Reply:
We should really remove configAsProperties https://github.com/apache/gobblin/blob/master/gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FlowSpec.java#L95
or change configAsProperties to executionLevelProperties which are updated, while the core configs which remain the same remain in `config`